src/org/opendatakit/briefcase/ui/reused/SourceConfigurationPanel.java

```diff
@@ -13,10 +13,21 @@
   private final List<Consumer<AggregateServerConnectionConfiguration>> onAggregateSourceCallbacks = new ArrayList<>();
   private final List<Consumer<Path>> onCustomDirectorySourceCallbacks = new ArrayList<>();
   private final List<Runnable> onResetCallbacks = new ArrayList<>();
+  public static final String PULL = "pull";
+  public static final String PUSH = "push";
+
+  public SourceConfigurationPanel(String action) {
```

Comment:
This is good!

Putting aside that passing a string feels arbitrary (maybe better an enum for that), constructors should be much simpler. 

This kind of conditional construction logic would go better into a pair of factories. You could have one factory for push and another for pull. Then, the factories could pass to the constructor both views.